our_site/planner/views.py

```diff
@@ -81,28 +81,95 @@ def dashboard(request):
             "MyEventsButton": DashboardFilterMyEvents(),
             "InvitedEventsButton": DashboardFilterInvitedEvents(),
-            "event_list" : event_list,
-            "owner" : owner,
+            "event_list": event_list,
+            "owner": owner,
         },
         request
     ))
 
 
-class EventCreationForm(forms.Form):
+class EventBasicDetails(forms.Form):
     event_name = forms.CharField(label="Event Name:", max_length=30, required=True)
-    event_date = forms.DateTimeField(input_formats=['%d/%m/%Y %H:%M'], label="Event Date (dd/mm/yy hh:mm):", required=True)
+    event_date = forms.DateTimeField(input_formats=['%d/%m/%Y %H:%M'], label="Event Date (dd/mm/yy hh:mm):",
+                                     required=True)
     event_description = forms.CharField(label="Event Description:", max_length=200, required=True)
     event_location = forms.CharField(label="Event Location:", max_length=50, required=True)
-    event_invitees_emails = forms.CharField(label="Invitees (comma delimited emails):", required=True)
-    event_roles = forms.CharField(label="Roles (comma separated zipped list of roles and quantities):", required=True)
 
 
 def event_creation(request):
     if request.user.is_anonymous:  # User is redirected to log in if they are not logged in
         return HttpResponseRedirect("/account/login/")
 
     if request.method == "GET":
-        template = loader.get_template('planner/event_creation.html')
-        form = EventCreationForm()
-        return HttpResponse(template.render({"form": form}, request))
+        template = loader.get_template("planner/event_creation.html")
+        event_creation_form = EventBasicDetails()
+        return HttpResponse(template.render({"event_creation_form": event_creation_form}, request))
+    elif request.method == "POST":
+        # It is acceptable to shove the data from this post into one form because currently only one form
+        # leads to this page
+        event_creation_form = EventBasicDetails(request.POST)
+        # If the form is valid, save the new event and move to it's editing page
+        # otherwise, just redisplay the form with its errors
+        if event_creation_form.is_valid():
+            event = Event(
+                owner=request.user.profile,
+                name=event_creation_form.cleaned_data["event_name"],
+                date=event_creation_form.cleaned_data["event_date"],
+                description=event_creation_form.cleaned_data["event_description"],
+                location=event_creation_form.cleaned_data["event_location"],
+            )
+            event.save()
+            return HttpResponseRedirect("/planner/{:d}/edit_event".format(event.id))
+        else:
+            template = loader.get_template("planner/event_creation.html")
+            return HttpResponse(template.render({"event_creation_form": event_creation_form}, request))
+
+
+def edit_event(request, event_id):
```

Comment:
Should we allow the option of deleting an event under edit_event or should that a completely different page they are lead to if they wanted to delete an event?

Reply:
This seems a discussion perhaps more pertinent to the organization of our issues rather than the code review itself. That was not included in the issue this fulfills (although I think it would be a good addition to this page). Is there a reason within the scope of the issues this pr closes that this influences how this code should be judged?

Reply:
No, there isn't, I looked at everything and everything looks good, just a quick question.